providers/class.two-factor-email.php

```diff
@@ -120,7 +120,10 @@ public function authentication_page( $user ) {
 			return;
 		}
 
-		$this->generate_and_email_token( $user );
+		if ( ! isset( $_GET['action'] ) || 'validate_2fa' !== $_GET['action'] ) {
```

Comment:
@joshbetz We should probably ensure that a valid token has been generated and is available for `validate_token()` before bailing here.